### PR TITLE
Support batch index ack feature

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolongran <xiaolongran@tencent.com>

Master Issue: #894

### Motivation

As #894 said, support batch index ack feature

### Modifications

- In ConsumerOptions, export `BatchIndexAcknowledgmentEnable` option, default: false.

### CODE REVIEW

File: pulsar/consumer_partition.go [hunk 1]
```diff
@@ -733,6 +734,7 @@ func (pc *partitionConsumer) internalAck(req *ackRequest) {
 
 func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, headersAndPayload internal.Buffer) error {
 	pbMsgID := response.GetMessageId()
+	ackSet := response.GetAckSet()
```
Comment (panszobe) [inline pulsar/consumer_partition.go:737]: It seems that ackSet not handled to judge which msg is acked.
It maybe cause duplicated consuming.
Would you please have a look, thanks a lot.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
Please add related unit tests

File: pulsar/impl_message.go [hunk 2]
```diff
@@ -350,17 +350,17 @@ func (msg *message) BrokerPublishTime() *time.Time {
 	return msg.brokerPublishTime
 }
 
-func newAckTracker(size int) *ackTracker {
+func newAckTracker(batchMsgSize int) *ackTracker {
```
Comment (BewareMyPower) [inline pulsar/impl_message.go:353]: It looks like the renaming is meaningless.

File: pulsar/consumer_partition.go (hunk 1 above)
Comment (BewareMyPower) [reply pulsar/consumer_partition.go:737]: The `ackSet` should be parsed into a `BitSet` to check if the messages should be skipped when constructing the messages from a batch.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
After the private chat with @wolfstudy, I will take over this task (#894). Close this PR first. /cc @hangc0276

**Comment by panszobe** [comment]
@BewareMyPower When will the task be completed?

**Comment by BewareMyPower** [comment]
I cannot guarantee the specific deadline, but it should not take much time. Maybe this or next week.

**Pull request closed without merging**